libraries/AP_InertialSensor/AP_InertialSensor_Backend.h

```diff
@@ -35,6 +35,41 @@ class DataFlash_Class;
 class AP_InertialSensor_Backend
 {
 public:
+    /**
+     * Data sampling mode. It tells the backend which method to use for
+     * sampling data. Some backends might not implement all modes.
+     */
+    enum SamplingMode {
+        /**
+         * Use the sensor's internal FIFO.
+         */
+        FIFO = 0,
+        /**
+         * Check if there's a new sample available and read it.
+         */
+        DATA_READY,
+    } mode;
+
```

Comment:
As we talked in person yesterday: this doesn't make much sense. Neither for bmi160 or others. Data ready pin is orthogonal to having  a FIFO. FIFO allows us to raise the sampling rate without raising the frequency of the sampling thread: we will get more than one sample at a time. Data ready pin is just a way to check there's data available without needing to read a register, which is much more expensive.

There's another  use of the data ready pin that is waiting for an interrupt on it and reading the samples when that happens. We support that in kernel drivers, but  not in userspace ones